js/treetest.js

```diff
@@ -154,54 +129,54 @@ var socket = {
 		this._socket.on('disconnect', (reason) => {
 			console.debug(`Socket disconnected. Reason: ${reason}`);
 		});
-	},
+	},	
 
 	emitPageLoadEvent: function() {
-		this._emitEvent({}, 'page load');
+		this._emitEvent({}, 'page_load');
 	},
 
-	emitActivateNodeEvent: function(node) {
-		const data = {
-			node: $('#tree').jstree().get_path(node),
-			currentTaskIndex: tasks.idx + 1,
-		};
-		this._emitEvent(data, 'activate node');
+	emitSelectNodeEvent: function(node) {
+		this._emitNodeActionEvent(node, 'select_node');
 	},
 
 	emitOpenNodeEvent: function(node) {
-		const data = {
-			node: $('#tree').jstree().get_path(node),
-			currentTaskIndex: tasks.idx + 1,
-		};
-		this._emitEvent(data, 'open node');
+		this._emitNodeActionEvent(node, 'open_node');
 	},
 
 	emitCloseNodeEvent: function(node) {
-		const data = {
-			node: $('#tree').jstree().get_path(node),
-			currentTaskIndex: tasks.idx + 1,
-		};
-		this._emitEvent(data, 'close node');
+		this._emitNodeActionEvent(node, 'close_node');
 	},
 
-	emitTaskChangedEvent: function(newTaskIndex) {
-		this._emitEvent({ newTaskIndex }, 'task changed');
+	emitTaskChangedEvent: function() {
+		this._emitEvent({}, 'task_changed');
 	},
 
-	emitSubmitResponseEvent: function(answers) {
-		this._emitEvent({ answers }, 'submit response');
+	emitSubmitResponseEvent: function() {
+		this._emitEvent({}, 'submit_response');
 	},
 
 	emitWindowVisibilityChangedEvent: function(newVisibilityState) {
-		const data = { newVisibilityState };
-		this._emitEvent(data, 'window visibility changed');
+		//https://html.spec.whatwg.org/multipage/interaction.html#visibility-state
+		const isVisible = newVisibilityState == 'visible' ? true : false
```

Comment:
```suggestion
		const isVisible = newVisibilityState === 'visible';
```

* The result of a comparison is already a boolean value
* `===` is good practice in JavaScript, because `==` tries to coerce types first and can have unexpected results